app/src/main/java/org/gdg/frisbee/android/arrow/ArrowActivity.java

```diff
@@ -249,65 +250,107 @@ private void score(final String id) {
             return;
         }
 
-        AppStateManager.load(getGoogleApiClient(), Const.ARROW_DONE_STATE_KEY).setResultCallback(new ResultCallback<AppStateManager.StateResult>() {
-            @Override
-            public void onResult(AppStateManager.StateResult stateResult) {
-                AppStateManager.StateConflictResult conflictResult = stateResult.getConflictResult();
-                AppStateManager.StateLoadedResult loadedResult = stateResult.getLoadedResult();
-
-                if (loadedResult != null) {
-                    final int statusCode = loadedResult.getStatus().getStatusCode();
-                    if (statusCode == AppStateStatusCodes.STATUS_OK
-                            || statusCode == AppStateStatusCodes.STATUS_STATE_KEY_NOT_FOUND) {
-                        previous = "";
-
-                        if (statusCode == AppStateStatusCodes.STATUS_OK) {
-                            previous = new String(loadedResult.getLocalData());
-
-                            if (previous.contains(id)) {
-                                Toast.makeText(ArrowActivity.this, R.string.arrow_already_tagged, Toast.LENGTH_LONG).show();
-                            } else {
-                                addTaggedPersonToCloudSave(id);
-                            }
-                        } else {
-                            addTaggedPersonToCloudSave(id);
-                        }
-                    }
-                } else if (conflictResult != null) {
-                    previous = mergeIds(new String(conflictResult.getLocalData()), new String(conflictResult.getServerData()));
+        new StoreSnapshotTask(getGoogleApiClient()).execute(id);
```

Comment:
All the Snapshot code made sync, and moved to AsyncTask. That's what the documentation suggests. We may even move this to a IntentService actually. The problem with that is that we have to create GoogleApiClient in the Service too.